Client/Ecosia/UI/WhatsNew/DataProvider/WhatsNewJSONDataProvider.swift

```diff
@@ -0,0 +1,38 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Core
+
+struct WhatsNewJSONDataProvider: WhatsNewDataProvider {
+    
+    let jsonData: Data?
+    private struct WhatsNewItemRoot: Decodable {
+        let items: [RemoteWhatsNewItem]
+    }
+
+    private struct RemoteWhatsNewItem: Decodable {
+        let imageURL: URL
+        let title: String
+        let subtitle: String
+    }
+
+    func getData() throws -> [WhatsNewItem] {
```

Comment:
I guess the version would also need to somehow be fetched and compared here to decide which items to return... Or are we assuming that whoever provides the json already has some context and filter the items out before sending?

Still, very minor since this is not currently used and only an example of how it can be in the future.

Reply:
Yeah, It’s indeed only for the example proposed - will remove it as part of a separate commit so once we have any JSON support, we’ll add it back and enrich it with the version check context!